### PR TITLE
Scene view camera controller app

### PR DESCRIPTION
**Pull request by changanxian**
Adds a microapp to demonstrate the usage of CameraController in the SceneView.

### CODE REVIEW

**Review by hud10837 — APPROVED** [review]
Looks good to me and works as expected 👍

**Review by puneet-pdx — COMMENTED** [review]
@changanxian some comments. looks good overall 👍🏽

**Comment by changanxian** [comment]
@puneet-pdx updated, have a review.

File: microapps/SceneViewCameraControllerApp/app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  ~  Unless required by applicable law or agreed to in writing, software
+  ~  distributed under the License is distributed on an "AS IS" BASIS,
+  ~  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~  See the License for the specific language governing permissions and
+  ~  limitations under the License.
+  ~
+  -->
+
+<resources>
+    <string name="app_name">SceneViewTestApp</string>
```
Comment (puneet-pdx) [inline microapps/SceneViewCameraControllerApp/app/src/main/res/values/strings.xml:20]:
```suggestion
    <string name="app_name">SceneViewCameraControllerApp</string>
```

File: microapps/SceneViewCameraControllerApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewcameracontrollerapp/ui/theme/Theme.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    onPrimary = Color.White,
+    onSecondary = Color.White,
+    onTertiary = Color.White,
+    onBackground = Color(0xFF1C1B1F),
+    onSurface = Color(0xFF1C1B1F),
+    */
+)
+
+@Composable
+fun SceneViewTestAppTheme(
```
Comment (puneet-pdx) [inline microapps/SceneViewCameraControllerApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewcameracontrollerapp/ui/theme/Theme.kt:59]:
```suggestion
fun SceneViewCameraControllerAppTheme(
```

File: microapps/SceneViewCameraControllerApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewcameracontrollerapp/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import com.arcgismaps.toolkit.sceneviewcameracontrollerapp.screens.MainScreen
+import com.arcgismaps.toolkit.sceneviewcameracontrollerapp.ui.theme.SceneViewTestAppTheme
+
+class MainActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        ArcGISEnvironment.apiKey =
+            ApiKey.create(BuildConfig.API_KEY)
+        setContent {
+            SceneViewTestAppTheme {
```
Comment (puneet-pdx) [inline microapps/SceneViewCameraControllerApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewcameracontrollerapp/MainActivity.kt:37]:
```suggestion
            SceneViewCameraControllerAppTheme {
```

File: microapps/SceneViewCameraControllerApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewcameracontrollerapp/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                SceneViewTestApp()
```
Comment (puneet-pdx) [inline microapps/SceneViewCameraControllerApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewcameracontrollerapp/MainActivity.kt:38]:
```suggestion
                SceneViewCameraControllerApp()
```

File: microapps/SceneViewCameraControllerApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewcameracontrollerapp/MainActivity.kt [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            }
+        }
+    }
+}
+
+@Composable
+fun SceneViewTestApp() {
```
Comment (puneet-pdx) [inline microapps/SceneViewCameraControllerApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewcameracontrollerapp/MainActivity.kt:45]:
```suggestion
fun SceneViewCameraControllerApp() {
```

File: microapps/SceneViewCameraControllerApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewcameracontrollerapp/MainActivity.kt [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    MainScreen()
+}
+
+@Preview(showBackground = true)
+@Composable
+fun AppPreview() {
+    SceneViewTestAppTheme {
```
Comment (puneet-pdx) [inline microapps/SceneViewCameraControllerApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewcameracontrollerapp/MainActivity.kt:52]:
```suggestion
    SceneViewCameraControllerApp {
```

File: microapps/SceneViewCameraControllerApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewcameracontrollerapp/MainActivity.kt [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+        SceneViewTestApp()
```
Comment (puneet-pdx) [inline microapps/SceneViewCameraControllerApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewcameracontrollerapp/MainActivity.kt:53]:
```suggestion
        SceneViewCameraControllerApp()
```

File: microapps/SceneViewCameraControllerApp/app/src/main/AndroidManifest.xml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+
+    <application
+        android:allowBackup="true"
+        android:dataExtractionRules="@xml/data_extraction_rules"
+        android:fullBackupContent="@xml/backup_rules"
+        android:icon="@mipmap/ic_launcher"
+        android:label="@string/app_name"
+        android:roundIcon="@mipmap/ic_launcher_round"
+        android:supportsRtl="true"
+        android:theme="@style/Theme.SceneViewTestApp"
```
Comment (puneet-pdx) [inline microapps/SceneViewCameraControllerApp/app/src/main/AndroidManifest.xml:33]:
```suggestion
        android:theme="@style/Theme.SceneViewCameraControllerApp"
```

File: microapps/SceneViewCameraControllerApp/app/src/main/AndroidManifest.xml [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+        tools:targetApi="31">
+        <activity
+            android:name="com.arcgismaps.toolkit.sceneviewcameracontrollerapp.MainActivity"
+            android:exported="true"
+            android:theme="@style/Theme.SceneViewTestApp">
```
Comment (puneet-pdx) [inline microapps/SceneViewCameraControllerApp/app/src/main/AndroidManifest.xml:38]:
```suggestion
            android:theme="@style/Theme.SceneViewCameraControllerApp">
```

File: microapps/SceneViewCameraControllerApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewcameracontrollerapp/MainActivity.kt [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+import android.os.Bundle
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.tooling.preview.Preview
+import com.arcgismaps.ApiKey
+import com.arcgismaps.ArcGISEnvironment
+import com.arcgismaps.toolkit.sceneviewcameracontrollerapp.screens.MainScreen
+import com.arcgismaps.toolkit.sceneviewcameracontrollerapp.ui.theme.SceneViewTestAppTheme
```
Comment (puneet-pdx) [inline microapps/SceneViewCameraControllerApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewcameracontrollerapp/MainActivity.kt:29]:
```suggestion
import com.arcgismaps.toolkit.sceneviewcameracontrollerapp.ui.theme.SceneViewCameraControllerAppTheme
```

File: microapps/SceneViewCameraControllerApp/app/src/main/res/values/themes.xml [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+  ~  distributed under the License is distributed on an "AS IS" BASIS,
+  ~  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~  See the License for the specific language governing permissions and
+  ~  limitations under the License.
+  ~
+  -->
+
+<resources>
+
+    <style name="Theme.SceneViewTestApp" parent="android:Theme.Material.Light.NoActionBar" />
```
Comment (puneet-pdx) [inline microapps/SceneViewCameraControllerApp/app/src/main/res/values/themes.xml:22]:
```suggestion
    <style name="Theme.SceneViewCameraControllerApp" parent="android:Theme.Material.Light.NoActionBar" />
```

File: microapps/SceneViewCameraControllerApp/app/src/main/res/values/themes.xml [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+  ~  Unless required by applicable law or agreed to in writing, software
+  ~  distributed under the License is distributed on an "AS IS" BASIS,
+  ~  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~  See the License for the specific language governing permissions and
+  ~  limitations under the License.
+  ~
+  -->
+
+<resources>
+
```
Comment (puneet-pdx) [inline microapps/SceneViewCameraControllerApp/app/src/main/res/values/themes.xml:21]: can remove this newline

**Review by puneet-pdx — COMMENTED** [review]
@changanxian there are still some references to SceneViewTestApp that I missed to point earlier.

**Comment by changanxian** [comment]
@puneet-pdx Updated, have another look, Thanks.

**Review by puneet-pdx — APPROVED** [review]

**Pull request merged**